components/apimgt/org.wso2.carbon.apimgt.eventing/src/test/java/org/wso2/carbon/apimgt/eventing/EventHubEventPublisherTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.eventing;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.carbon.databridge.commons.Event;
+
+/**
+ * Test case for EventHubEventPublisher
+ */
+public class EventHubEventPublisherTest {
+    @Test
+    public void publishEventToEventHub() {
+        try {
+            EventPublisherFactory eventPublisherFactory = EventPublisherFactoryProducer
+                    .getEventPublisherFactory("org.wso2.carbon.apimgt.eventing.EventHubEventPublisherFactory");
+            EventPublisher eventPublisher = eventPublisherFactory.getEventPublisher();
+            eventPublisher.init();
+            Event event = new Event("org.wso2.apimgt.example.stream:1.0.0", System.currentTimeMillis(),
+                    null, null, null);
+//            eventPublisher.publish(event);
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            Assert.fail("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Couldn't instantiate EventPublisherFactory.");
+        }
+    }
+}
```

Comment:
Usually we woul write one 'EventPublisherFactoryProducerTest' to test instanceOf to assert that the Producer gives the correct instance without any error and have another 'EventHubEventPublisherTest' to assert init(not required in this because it does nothhing) and 'publish' upon creating a new EventHubEventPublisher object.